SalesforceSDK/Salesforce.SDK.App/Auth/PincodeDialog.xaml.cs

```diff
@@ -63,14 +63,20 @@ public PincodeDialog()
 
         protected override void OnNavigatedTo(NavigationEventArgs e)
         {
-            if (e.Parameter != null && !(e.Parameter is PincodeOptions))
+            // start with a default PincodeOptions object
+            Options = new PincodeOptions(PincodeOptions.PincodeScreen.Create, AccountManager.GetAccount(), "");
+            if (e.Parameter != null)
```

Comment:
checking a type is less expensive than catching an exception, so I would still put a check in this if statement to make sure e.Parameter is PincodeOptions... like this:

if(e.Parameter!=null && e.Parameter is PincodeOptions)